modules/tensor_mechanics/src/interfacekernels/CZMInterfaceKernel.C

```diff
@@ -0,0 +1,222 @@
+                       "displacement to couple on the master side. "
+                       "If disp_index == 0, then disp_1 = disp_y "
+                       "If disp_index == 1, then disp_1 = disp_x "
+                       "If disp_index == 2, then disp_1 = disp_x");
+  params.addCoupledVar("disp_1_neighbor",
+                       "Name of the variable representing "
+                       "the 2nd displacement to couple on the slave side. "
+                       "If disp_index == 0, disp_1_neighbor = disp_y_neighbor "
+                       "If disp_index == 1, disp_1_neighbor = disp_x_neighbor "
+                       "If disp_index == 2, disp_1_neighbor = disp_x_neighbor");
+  params.addCoupledVar("disp_2",
+                       "Name of the variable representing the 3rd "
+                       "displacement to couple on the master side. "
+                       "If disp_index == 0, then disp_2 = disp_z "
+                       "If disp_index == 1, then disp_2 = disp_z "
+                       "If disp_index == 2, then disp_2 = disp_y ");
+  params.addCoupledVar("disp_2_neighbor",
+                       "Name of the variable representing "
+                       "the 3rd displacement to couple on the slave side. "
+                       "If disp_index == 0, disp_2_neighbor = disp_z_neighbor "
+                       "If disp_index == 1, disp_2_neighbor = disp_z_neighbor "
+                       "If disp_index == 2, disp_2_neighbor = disp_y_neighbor");
+  params.addParam<std::string>(
+      "residual",
+      "traction",
+      "The name of the material property representing the residual coefficients");
+  params.addParam<std::string>(
+      "jacobian",
+      "traction_spatial_derivatives",
+      "The name of the  material property representing the jacobian coefficients");
+  params.addClassDescription("Cohesive Zone Interface Kernel for non-stateful"
+                             "cohesive laws depending only on the displacement Jump");
+
+  return params;
+}
+
+CZMInterfaceKernel::CZMInterfaceKernel(const InputParameters & parameters)
+  : InterfaceKernel(parameters),
+    _disp_index(getParam<unsigned int>("disp_index")),
+    _disp_1(_mesh.dimension() >= 2 ? coupledValue("disp_1") : _zero),
+    _disp_1_neighbor(_mesh.dimension() >= 2 ? coupledNeighborValue("disp_1_neighbor") : _zero),
+    _disp_2(_mesh.dimension() >= 3 ? coupledValue("disp_2") : _zero),
+    _disp_2_neighbor(_mesh.dimension() >= 3 ? coupledNeighborValue("disp_2_neighbor") : _zero),
+    _disp_1_var(coupled("disp_1")),
+    _disp_1_neighbor_var(coupled("disp_1_neighbor")),
+    _disp_2_var(coupled("disp_2")),
+    _disp_2_neighbor_var(coupled("disp_2_neighbor")),
+    // residual and jacobian coefficients are material properties and represents
+    // the residual and jacobain of the traction sepration law wrt the displacement jump.
+    _residual(getParam<std::string>("residual")),
+    _jacobian(getParam<std::string>("jacobian")),
```

Comment:
You do not need string of `_residual` and `_jacobian`. You can do getMaterialProperty<RealVectorValue>(getParam<std::string>("residual")), etc.